simpleviewinc/keg-cli
# Vis 976 added injected apps

**Ticket**: [VIS-976](https://jira.simpleviewtools.com/browse/VIS-976)

## Context
* This update allows external apps to be injected into the keg-cli
* Visitapps back setup is being updated to use a new setup
  * This update will allow us to run Visitapps within the keg-cli
* A couple minor changes to the install scripts to allow installing on Windows WSL
  * This has been tested with Nicole

## Goal
* Run external apps within the Keg-CLI
* Install the Keg-CLI on Windows WSL

## Updates
* Due to how external apps are injected into the Keg-CLI, the `keg docker package` task must be run differently
  * New package tasks have been added to `taps`, which work the same way at the original `package` commands
  * To call them like you call any other tap task
	* `keg <tap-name> package run --package <options>`
* Fixed the `keg components stop` task
  * `stop` task was not added to the root `components` task
* Number of clean up items
  * Cleaned up some git tasks
  * Cleaned up calling `convertParamsToEnvs` twice in some cases
* Added a few random tasks
  * Added a `docker volumes` list task
* Added jsDocs in a few locations
* Updated some key names for consistency
* Normalized how services build their arguments with `overrides` and `__injected` apps
* Added an `inject` key to tap tasks
  * This indicates that the task can be injected by a external app 

## Testing

**Test 1**
* Run an injected app with pacakge task
  * `keg vab pack run --package docker.pkg.github.com/simpleviewinc/g-packages/va-backend:test-package-branch`
	* Should run the app
* Run an injected app with package run task
  * `keg vab pack run --package docker.pkg.github.com/simpleviewinc/keg-packages/va-backend:test-package-branch --command ls`
    * Should print out the current working directory
* Test the non-injected packages
  * `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-core:zen-229-core-deps`
* Test with a command
  * `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-core:zen-229-core-deps --command test`

**Test 2**
* Clone this repo => [Keg-External-Example](https://github.com/simpleviewinc/keg-external-example)
  * Follow the instructions in the `README.md`
